content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,457 @@
+
+## Signaling
+
+After registration, 
+the users can participate in the application by sending signals to the other participants in a decentralised manner or to a centralised server. 
+Along with their signal, 
+they need to generate a ZK-Proof by using the circuit with the specification described above. 
+
+For generating a proof, 
+the users need to obtain the required parameters or compute them themselves, 
+depending on the application implementation and client libraries supported by the application.
+For example the users can store the memebrship merkle tree on their end and 
+generate a merkle proof whenever they want to generate a signal. 
+
+### Signal hash
+
+The signal hash is generated by hashing the raw signal (or content) using the `keccak256` hash function, 
+converting the hashed value to number and padding it:
+
+```
+  signal: string = "signal"
+  converted: string = toHex(toUtf8Bytes(signal))
+  signal_hash: bigint = BigInt(keccak256(converted)) >> BigInt(8)
+```
+
+
+### Epoch
+
+The epoch (or external nullifier) is also obtained by hasing a raw string value using `keccak256`:
+
+```
+    raw_epoch: string = "1234"
+    converted: string = keccak256(raw_epoch)
+    hex_string: string = converted.slice(8, 64)
+    epoch: string = "0x" + hex_string.padStart(64, "0")
+```
+
+
+### Merkle proof
+
+The merkle proof should be obtained locally or from a trusted third party. 
+By using the incremental merkle tree algorithm, 
+the merkle can be obtained by providing the `leaf_index` of the `identity_commitment`. 
+The proof (`merkle_proof`) is composed of the following fields:
+
+```
+{
+  root: bigint
+  indices: number[]
+  path_elements: bigint[][]
+}
+```
+
+
+### Generating proof
+
+For proof generation, 
+the user need to submit the following fields to the circuit:
+
+```
+    {
+      identity_secret: identity_secret_hash,
+      path_elements: merkle_proof.path_elements,
+      identity_path_index: merkle_proof.indices,
+      x: signal_hash,
+      epoch: epoch,
+      rln_identifier: rlnIdentifier
+    }
+```
+
+The generated proof (`zk_proof`) is composed of the following fields:
+
+```
+{
+  pi_a: Array<string>,
+  pi_b: [ [Array<string>], [Array<string>], [Array<string>] ],
+  pi_c: Array<string>
+  protocol: string,
+  curve: string
+}
+```
+
+### Calculating output
+
+The proof output is calculated locally, 
+in order for the required fields for proof verification to be sent along with the proof. 
+The proof output is composed of the `y` share of the secret equation and the `nullifier`. 
+The following fields are needed for proof output calculation:
+
+```
+{
+  identity_secret: bigint, 
+  epoch: bigint, 
+  x: bigint, 
+  limit: number, 
+  rln_identifier: bigint
+}
+```
+
+The output `[y, nullifier]` is calculated in the following way:
+
+```
+a_0 = identity_secret
+a_1 = poseidonHash([a0, epoch])
+
+y = a_0 + x * a_1
+
+nullifier = poseidonHash([a_1, rln_identifier])
+```
+
+
+
+### Sending proof
```

Comment:
This head makes it seem like it is just sending proof, but it is sending "output message", which includes proof

Reply:
Yes, will be changed.